hsds/datanode.py

```diff
@@ -165,8 +167,16 @@ async def bucketScan(app):
                 tb = traceback.format_exc()
                 print("traceback:", tb)
 
-        log.info(f"bucketScan - sleep: {async_sleep_time}")
-        await asyncio.sleep(async_sleep_time)
+            last_action = time.time()
+
+        now = time.time()
+        if (now - last_action) > async_sleep_time:
+            sleep_time = async_sleep_time  # long nap
+        else:
+            sleep_time = short_sleep_time  # shot nap
```

Comment:
What's the idea behind varying the time between scans based on how long the scan itself takes? Or is `last_action` tracking something else?

Reply:
If there's nothing going on, use a longer sleep and save cpu cycles.  It's not how long the scan takes but rather the last time there was a domain to scan.

Reply:
Here's a quick check using the value_test.py test case.
I modified the testPut1DDataset to do a sequence of flushes and rescans:

        # check values we should get from a verbose query
        import time
        ts = time.time()
        for i in range(10):
            expected = {"num_chunks": 1, "allocated_size": 40}
            self.checkVerbose(dset_id, headers=headers, expected=expected)
            now = time.time()
            elapsed = now - ts
            print(f"flush {elapsed:.2f}")
            ts = now

Output is:

          $ python value_test.py ValueTest.testPut1DDataset
          testPut1DDataset /home/test_user1/hsds_test/valuetest/20231010T154859_160988Z
          flush 2.54
          flush 0.12
          flush 0.12
          flush 0.13
          flush 0.13
          flush 0.13
          flush 0.12
          flush 0.12
          flush 0.13
          flush 0.13


The first round takes a couple of seconds because apparently  the bucketScan task is in long sleep mode.  After the first flush, it "wakes up" and each update only takes about 0.1s.